src/expand_quote_removal.c

```diff
@@ -0,0 +1,80 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   expand_quote_removal.c                             :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: gwolf <gwolf@student.42vienna.com>         +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/08/03 07:49:12 by gwolf             #+#    #+#             */
+/*   Updated: 2023/08/10 11:57:17 by gwolf            ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+/**
+ * @file expand_quote_removal.c
+ * @brief Functions for quote removal.
+ */
+#include "mod_expand.h"
+
+/**
+ * @brief Removes quotes from a string, ignores quoted quotes.
+ *
+ * @param str Where to remove quotes.
+ * @return t_err SUCCESS
+ */
+t_err	ft_quote_removal(char *str)
+{
+	t_track	input;
+	t_err	err;
+
+	ft_init_tracker(&input, str);
+	while (input.str[input.pos])
+	{
+		if (input.str[input.pos] == '\'' && !input.quoted)
+			err = ft_rm_single_quote(&input);
+		else if (input.str[input.pos] == '"')
+			err = ft_rm_double_quote(&input);
+		else
+			err = ft_move_tracker(&input);
+		if (err != SUCCESS)
```

Comment:
Removed return value of ft_quote_removal()

Reply:
Update: removed whole functions since nowhere used anymore :D

Reply:
Update2: deleted file expand_quote_removal.c by putting it's functions into the expand_expander.c since this is the main function which calls the quote removal functions.